include/ignition/gazebo/EntityComponentManager.hh

```diff
@@ -91,6 +91,35 @@ namespace ignition
       public: void RequestRemoveEntity(const Entity _entity,
           bool _recursive = true);
 
+      /// \brief Prevent an entity and optionally its children from
+      /// being removed.
```

Comment:
The word "lock" implies to me that more is being prevented than just removal. Without reading these docs, I assumed the function would prevent the entity's components from being modified.

How about something like `MarkAsUnremovable`?

Reply:
Changed the name in 1e908d310e46dfd94c1529c1a658ac8ec09397df.